mmda/types/annotation.py

```diff
@@ -125,7 +123,8 @@ def find(self, query: Annotation) -> List[Annotation]:
 @dataclass
 class SpanGroupIndexer(Indexer):
 
-    _index: IntervalTree = IntervalTree()
+    # careful; if write it as _index = IntervalTree(), all SpanGroupIndexers will share the same _index object
```

Comment:
Oh yes that's true - and this is almost the same as the list problem. Sorry for missing it earlier!